source/server/http/admin.cc

```diff
@@ -1075,12 +1075,21 @@ std::string AdminImpl::runtimeAsJson(
 }
 
 Http::Code AdminImpl::handlerRuntimeModify(absl::string_view url, Http::HeaderMap&,
-                                           Buffer::Instance& response, AdminStream&) {
-  const Http::Utility::QueryParams params = Http::Utility::parseQueryString(url);
+                                           Buffer::Instance& response, AdminStream& admin_stream) {
+  Http::Utility::QueryParams params = Http::Utility::parseQueryString(url);
   if (params.empty()) {
-    response.add("usage: /runtime_modify?key1=value1&key2=value2&keyN=valueN\n");
-    response.add("use an empty value to remove a previously added override");
-    return Http::Code::BadRequest;
+    // Check if the params are in the request's body.
+    // TODO(rgs1): check the content-type header in the request.
+    if (admin_stream.getRequestBody() != nullptr) {
+      params = Http::Utility::parseFormBody(admin_stream.getRequestBody()->toString());
+    }
+
+    if (params.empty()) {
+      response.add("usage: /runtime_modify?key1=value1&key2=value2&keyN=valueN\n");
```

Comment:
The code basically looks fine, but I think it's a little weird to take the query-params from the body. I think they should probably be post params, as this is a POST (or should be, given that it is a write operation).

Reply:
Hmm what do you mean by post params? Isn't a query string in the body essentially post params:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST

Well, except for the fact that we are not enforcing `Content-Type: application/x-www-form-urlencoded`?

Reply:
I see...actually would it be a problem to enforce the content-type? Also, could you test -- at least manually -- with a curl POST?

Reply:
Curl test:

```
$ curl --location --request POST --data "foo1=bar1&foo2=bar2" "https://postman-echo.com/post"
{"args":{},"data":"","files":{},"form":{"foo1":"bar1","foo2":"bar2"},"headers":{"x-forwarded-proto":"https","host":"postman-echo.com","content-length":"19","accept":"*/*","content-type":"application/x-www-form-urlencoded","user-agent":"curl/7.54.0","x-forwarded-port":"443"},"json":{"foo1":"bar1","foo2":"bar2"},"url":"https://postman-echo.com/post"}
```

Yes, I can actually enforce the `content-type`.